includes/search.rs

```diff
@@ -738,76 +738,122 @@ pub(crate) fn filter_for_operator(
     operator: PredicateOperator,
 ) -> Result<(), String> {
     for object in unfiltered {
-        let val = match search_field {
-            "id" => &object[search_field],
+        let mut val = match search_field {
+            "id" => object[search_field].clone(),
             _ => {
                 let data = &object["data"];
-                &data[search_field]
+                data[search_field].clone()
             }
         };
 
+        let ids = &object["ids"];
+        println!(
+            "(filter_for_operator) (MOCK) test ids field [{}].",
+            ids,
+        );
+        if val == serde_json::json!(null) && *ids != serde_json::json!(null) {
+            println!(
+                "(filter_for_operator) (MOCK) found ids [{}].",
+                ids,
+            );
+            let ids: Vec<super::FieldValue> = serde_json::from_value(ids.clone()).map_err(|e| format!("Could not convert [{:?}] to Ids from json value: {}", ids, e))?;
+            println!("search ids after convert: {:?}", ids);
+            for id in ids {
```

Comment:
If you're searching for just the first instance (or only expect one instance), can use `find` which will stop iterating once it finds the first entry that satisfies the predicate:
```rust
ids
    .iter()
    .find(|id| id.field == search_field)
    .map(|id| val = serde_json::json!(id.value));
```

Also potentially don't need to shadow `ids`:
```rust
serde_json::from_value(ids.clone())
    .map_err(|e| {
       format!("Could not convert [{:?}] to Ids from json value: {}", ids, e)
    })?
    .iter()
    .find(|id| id.field == search_field)
    .map(|id| val = serde_json::json!(id.value));
```

Reply:
fixed: [chore: code style fixes](https://github.com/aetheric-oss/svc-storage/pull/105/commits/d05d63638158b993e616bd1cea105add3ec43535)